packages/penrose-core/src/engine/EngineUtils.ts

```diff
@@ -615,32 +642,32 @@ export const insertExpr = (
             // Deal with vector values
             const res2 = res.contents;
             if (res2.tag !== "VectorV") {
-              const err = "expected Vector";
               if (compiling) {
-                return addWarn(trans, err);
+                return addWarn(trans, runtimeValueTypeError(path, "Vector", res2.tag));
               }
+              const err = "expected Vector";
               throw Error(err);
```

Comment:
Did you miss this one?

Reply:
No, it's a consistent pattern in `insertExpr` that the error is returned if `compiling=true`, otherwise (at runtime) it makes an error that is thrown.